lib/absinthe/phase/schema.ex

```diff
@@ -286,20 +286,22 @@ defmodule Absinthe.Phase.Schema do
   defp determine_concrete_type(result, node, schema) do
     with %{type: type} <- result,
          %{input_value: %{normalized: %{fields: fields}}} <- node,
-         %Absinthe.Type.InputUnion{resolve_type: resolve_type} <-
+         %Absinthe.Type.InputUnion{} <-
            Absinthe.Schema.cached_lookup_type(schema, Type.unwrap(type)) do
-      concrete_type = evaluate_resolve_type(fields, resolve_type)
+      concrete_type = extract_typename(fields)
       %{result | type: concrete_type}
     else
       _ -> result
     end
   end
 
-  defp evaluate_resolve_type(fields, resolve_type) do
-    fields
-    |> Enum.into(%{}, fn
-      %{name: name, input_value: %{normalized: %{value: value}}} -> {name, value}
-    end)
-    |> resolve_type.()
+  defp extract_typename(fields) do
+    input_fields =
+      fields
+      |> Enum.into(%{}, fn
```

Comment:
probably don't need to do this conversion into a map anymore, and just get the `typename` directly